app/models/pdc_serialization/datacite.rb

```diff
@@ -0,0 +1,160 @@
+  #
+  # For testing purposes you can also quickly get an XML serialization with the helper:
+  #
+  #   datacite_xml = PDCSerialization::Datacite.skeleton_datacite_xml(...)
+  #
+  # You can also pass a PDCMetadata::Resource which is useful to test with a more
+  # complex dataset without saving the work to the database:
+  #
+  #   json = {...}.to_json
+  #   resource = PDCMetadata::Resource.new_from_json(json)
+  #   datacite_xml = PDCSerialization::Datacite.new_from_work_resource(resource).to_xml
+  #
+  # For information
+  #   Datacite schema: https://support.datacite.org/docs/datacite-metadata-schema-v44-properties-overview
+  #   Datacite mapping gem: https://github.com/CDLUC3/datacite-mapping
+  #
+  # rubocop:disable Metrics/ClassLength
+  class Datacite
+    attr_reader :mapping
+
+    def initialize(mapping)
+      @mapping = mapping
+    end
+
+    # Returns the XML serialization for the Datacite record
+    # Note that the actual XML serialization is done by the datacite-mapping gem.
+    def to_xml
+      @mapping.write_xml
+    end
+
+    ##
+    # Returns the XML serialization for a valid Datacite skeleton record based on a few required values.
+    # Useful for early in the workflow when we don't have much data yet and for testing.
+    #
+    # @param [String] identifier, e.g "10.1234/567"
+    # @param [String] title
+    # @param [String] creator
+    # @param [String] publisher
+    # @param [String] publication_year
+    # @param [String] resource_type
+    def self.skeleton_datacite_xml(identifier:, title:, creator:, publisher:, publication_year:, resource_type:)
+      mapping = ::Datacite::Mapping::Resource.new(
+        identifier: ::Datacite::Mapping::Identifier.new(value: identifier),
+        creators: [] << ::Datacite::Mapping::Creator.new(name: creator),
+        titles: [] << ::Datacite::Mapping::Title.new(value: title),
+        publisher: ::Datacite::Mapping::Publisher.new(value: publisher),
+        publication_year: publication_year,
+        resource_type: datacite_resource_type(resource_type)
+      )
+      mapping.write_xml
+    end
+
+    ##
+    # Creates a PDCSerialization::Datacite object from a Work
+    def self.new_from_work(work)
+      new_from_work_resource(work.resource)
+    end
+
+    ##
+    # Creates a PDCSerialization::Datacite object from a PDCMetadata::Resource
+    def self.new_from_work_resource(resource)
+      mapping = ::Datacite::Mapping::Resource.new(
+        identifier: ::Datacite::Mapping::Identifier.new(value: resource.doi),
+        creators: creators_from_work_resource(resource.creators),
+        titles: titles_from_work_resource(resource.titles),
+        publisher: ::Datacite::Mapping::Publisher.new(value: resource.publisher),
+        publication_year: resource.publication_year,
+        resource_type: datacite_resource_type(resource.resource_type)
+      )
+      Datacite.new(mapping)
+    end
+
+    class << self
+      def creators_from_work_resource(creators)
```

Comment:
@hectorcorrea This will need the private keyword to make them private.

Reply:
Are you sure? If I run this on the terminal the method is not available:

```
[1] pry(main)> datacite = PDCSerialization::Datacite.new({})
=> #<PDCSerialization::Datacite:0x00007fa16fcb35c8 @mapping={}>

[2] pry(main)> datacite.datacite_resource_type("x")
NoMethodError: undefined method `datacite_resource_type' for #<PDCSerialization::Datacite:0x00007fa16fcb35c8 @mapping={}>
from (pry):2:in `<main>'


```

Reply:
nevermind...my bad

Reply:
@carolyncole thanks for the catch. I've marked those methods as private. Now they are more private that [Magnum PI](https://en.wikipedia.org/wiki/Magnum,_P.I.) :)